File: app/src/main/java/com/example/brauliomendez/cleanarquitecturedaggerandgithubapi/search/domain/repository/GithubSearchRepository.java
package com.example.brauliomendez.cleanarquitecturedaggerandgithubapi.search.domain.repository;


import com.example.brauliomendez.cleanarquitecturedaggerandgithubapi.search.domain.entity.SearchResponse;

import io.reactivex.Observable;

/**
 * Created by Leo on 24/01/17.
 */

public interface GithubSearchRepository {
  Observable<SearchResponse> searchGithubRepositories(String username);
}

File: kotlinversion/src/main/java/com/example/kotlinversion/domain/repository/GithubSearchRepository.kt
package com.example.kotlinversion.domain.repository

import com.example.kotlinversion.domain.entity.SearchResponse
import io.reactivex.Observable

/**
 * Created by BraulioMendez on 1/29/17.
 */
interface GithubSearchRepository {

    fun searchGithubRepositories(username : String) : Observable<SearchResponse>
}